### PR TITLE
Cleanup and fix enumeration of USB devices

### PR DESCRIPTION
**Pull request by izacus**
The current USB code uses only vendor ID to identify USB devices and then immediately tries to open them. Unfortunately, Samsung and some others are also major manufacturers of USB flash drives, which means that headunit tries to run AA on a USB stick plugged into the car in some cases ;)

This PR fixes the enumeration and the code will now actually check if the device supports Android Accessory Protocol before assigning it priority. It'll also be more patient when waiting for the device on the bus.

Also cleans up the code around there a little bit.

**TL;DR:**  this fixes AA not working if you have a USB stick plugged in.

Tested on EU firmware 54.0.513 and Ubuntu.

### CODE REVIEW

**Comment by izacus** [comment]
Hmm, found a bug or two still, will reopen after fixing them.

**Comment by davepro305** [comment]
Do you plan to release a 0.95 version with all those fixes?
